exec/java-exec/src/main/java/org/apache/drill/exec/work/fragment/FragmentExecutor.java

```diff
@@ -231,9 +261,9 @@ public Void run() throws Exception {
           while (shouldContinue()) {
             // Fragment is not cancelled
 
-            if (eventProcessor.hasFinishedRequests()) {
+            if (!recieverFinishedQueue.isEmpty()) {
```

Comment:
I don't think `isEmpty()` is necessary, it is the same as `poll() != null`.

Reply:
Thanks for catching removed.